src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -171,9 +168,16 @@ export async function uploadArtifactToS3(fileName: string, resp: CreateUploadUrl
             'x-amz-checksum-sha256': sha256,
             'Content-Type': 'application/zip',
             'x-amz-server-side-encryption': 'aws:kms',
-            'x-amz-server-side-encryption-aws-kms-key-id': resp.kmsKeyArn,
+            'x-amz-server-side-encryption-aws-kms-key-id': kmsKeyArn,
         }
     }
+    return headersObj
+}
+
+// TODO: later, consider enhancing the S3 client to include this functionality
+export async function uploadArtifactToS3(fileName: string, resp: CreateUploadUrlResponse) {
+    const sha256 = getSha256(fileName)
+    const headersObj = getHeadersObj(sha256, resp.kmsKeyArn)
 
     await sleep(2000) // pause to give time to recognize potential cancellation
```

Comment:
These are everywhere. I don't understand why gumby is the only feature that requires sleep() calls all over the code.

Reply:
As mentioned, it's in my plans to re-evaluate the necessity of the sleep() calls